app/controllers/application_controller.rb

```diff
@@ -6,5 +6,15 @@ class ApplicationController < ActionController::Base
   def set_current_user
     Current.user = User.find_by(id: session[:user_id]) if session[:user_id]
   end
-  
+
+  def auth_check
+    # allows only logged in user
+    @user = User.find_by(id: session[:user_id])
+    if session[:user_id] != @user.id
```

Comment:
Is it even possible? I mean first you are find a user with `session[:user_id]` and then matching if the ids are equal or not. This should always be false.